public_html/wp-content/mu-plugins/virtual-embeds/virtual-embeds.php

```diff
@@ -78,3 +89,64 @@ function render_crowdcast_block( $attributes ) {
 
 	return '<div class="wp-block-wordcamp-crowdcast-embed ' . $classes . '">' . $embed . '</div>';
 }
+
+/**
+ * Render the YouTube Live Chat embed on the front end.
+ *
+ * @param array $attributes Block attributes.
+ *
+ * @return string
+ */
+function render_youtube_live_chat_block( $attributes ) {
+	$classes      = 'wp-block-wordcamp-youtube-live-chat-embed';
+	$embed_domain = wp_parse_url( site_url(), PHP_URL_HOST );
+	$query_parts  = wp_parse_args( wp_parse_url( $attributes['videoUrl'] ?? '', PHP_URL_QUERY ) );
+
+	if ( ! isset( $query_parts['v'] ) ) {
+		return '';
+	}
+
+	if ( ! empty( $attributes['align'] ) ) {
+		$classes .= ' align' . sanitize_html_class( $attributes['align'] );
+	}
+
+	$iframe_src = sprintf(
+		'https://www.youtube.com/live_chat?v=%s&embed_domain=%s',
+		$query_parts['v'],
+		$embed_domain
+	);
+
+	ob_start();
+	?>
+
+	<div class="<?php echo esc_attr( $classes ); ?>">
+		<iframe
+			id="wp-block-wordcamp-youtube-live-chat-embed__video-<?php echo esc_attr( $query_parts['v'] ); ?>"
+			title="Embedded YouTube live chat"
+			src="<?php echo esc_url( $iframe_src ); ?>"
+			sandbox="allow-same-origin allow-scripts allow-popups"
+		>
+		</iframe>
+
+		<?php
+		/*
+		 * Apparently YouTube artificially blocks this entire feature on phones and tablets, and only mentions
+		 * that in passing in their documentation.
+		 *
+		 * There don't seem to be any practical workarounds. See https://stackoverflow.com/a/12845483/450127
+		 * and https://stackoverflow.com/a/58739531/450127.
+		 *
+		 * So... yeah. We're left with this.
+		 */
+		?>
+
+		<p class="wp-block-wordcamp-youtube-live-chat-embed__boneheaded-decision">
```

Comment:
I'm not really comfortable with this being in our public-facing code, even if it is a very strange thing for youtube to do. What about this?

```suggestion
		<p class="wp-block-wordcamp-youtube-live-chat-embed__availability-warning">
```

Reply:
It seems pretty mild to me, and something they deserve to be publicly criticized for, but I don't feel strongly. I'll update the CSS reference too.